### PR TITLE
BUG: constructing a DataFrame using range doesn't work

### LINKED ISSUE #26342: Constructing DataFrames using range fails
Working on removing ``compat.lrange``, I've discovered that 1-dim lists, iterators and generators work as expected:

```python
>>> pd.DataFrame(list(range(3)))  # ok
   0
0  0
1  1
2  2
>>> pd.DataFrame(iter(range(3)))  # ok
   0
0  0
1  1
2  2
>>> pd.DataFrame((i for i in range(3))  # ok
   0
0  0
1  1
2  2
```

while constructing with a plain ``range`` does not work:

```python
>>> pd.DataFrame(range(3))  # not ok
AttributeError: 'range' object has no attribute 'ndim'
```

Fix upcoming.

### PR DESCRIPTION
**Pull request by topper-123**
- [x] closes #26342
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Fixes bug when constructing a ``DataFrame`` using a ``range``.

### CODE REVIEW

File: pandas/tests/frame/test_constructors.py [hunk 1]
```diff
@@ -1011,9 +1024,13 @@ def __iter__(self):
...
 
     def test_constructor_iterator(self):
+        result = DataFrame(iter(range(10)))
+        expected = DataFrame(list(range(10)))
+        tm.assert_frame_equal(result, expected)
 
+    def test_constructor_list_of_iterators(self):
+        result = DataFrame([iter(range(10)), iter(range(10))])
         expected = DataFrame([list(range(10)), list(range(10))])
-        result = DataFrame([range(10), range(10)])
```
Comment (topper-123) [inline pandas/tests/frame/test_constructors.py]: A plain ``range`` isn't a iterator, so this fixes up the nomenclature.

The range case is now placed in ``test_constructor_list_of_ranges``, which is clearer.

File: doc/source/whatsnew/v0.25.0.rst [hunk 2]
```diff
@@ -258,7 +258,10 @@ Performance Improvements
 Bug Fixes
 ~~~~~~~~~
 
+DataFrame
+^^^^^^^^^^^
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.25.0.rst]: Too many "^" (just enough to funderline `DataFrame`)

File: doc/source/whatsnew/v0.25.0.rst [hunk 3]
```diff
@@ -258,7 +258,10 @@ Performance Improvements
...
 
+- Fixed a bug where Constructing a :class:`DataFrame` using a ``range`` (e.g. ``pd.DataFrame(range(3))``) would cause an ``AttributeException`` (:issue:`26342`).
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.25.0.rst]: I think you meant ``AttributeError``

File: pandas/tests/frame/test_constructors.py [hunk 4]
```diff
@@ -999,6 +1001,17 @@ def __len__(self, n):
...
         tm.assert_frame_equal(result, expected, check_dtype=False)
 
+    def test_constructor_range(self):
+        # GH26342
+        result = DataFrame(range(10))
+        expected = DataFrame(list(range(10)))
+        tm.assert_frame_equal(result, expected)
+
+    def test_constructor_list_of_ranges(self):
+        result = DataFrame([range(10), range(10)])
```
Comment (gfyoung) [inline pandas/tests/frame/test_constructors.py]: I would reference the issue as a comment above this line as well.

**Review by gfyoung — APPROVED** [review]
Will need to wait on fixing the Windows build, but the patch looks good!

cc @jreback

**Comment by jreback** [comment]
i will comment later

File: doc/source/whatsnew/v0.25.0.rst [hunk 5]
```diff
@@ -258,7 +258,10 @@ Performance Improvements
 Bug Fixes
 ~~~~~~~~~
 
+DataFrame
```
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: just put this in the reshaping section, don't create a new one

File: doc/source/whatsnew/v0.25.0.rst (hunk 3 above)
Comment (jreback) [inline doc/source/whatsnew/v0.25.0.rst]: Bug in construction of a ``:class:`DataFrame` `` using a ``range`` object (:issue:....)

no need to be more verbose; we rarely mention the original error received

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/test_constructors.py (hunk 4 above)
Comment (topper-123) [reply pandas/tests/frame/test_constructors.py]: Oh, this is just an existing test that has been moved. It already passed before this PR.

**Comment by topper-123** [comment]
This passes now.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @topper-123

**Pull request merged**